snow/engine/avalanche/transitive.go

```diff
@@ -500,38 +498,43 @@ func (t *Transitive) issue(vtx avalanche.Vertex) error {
 // Otherwise, some txs may not be put into vertices that are issued.
 // If [empty], will always result in a new poll.
 func (t *Transitive) batch(txs []snowstorm.Tx, force, empty bool) error {
-	batch := make([]snowstorm.Tx, 0, t.Params.BatchSize)
 	issuedTxs := ids.Set{}
 	consumed := ids.Set{}
 	issued := false
 	orphans := t.Consensus.Orphans()
-	for _, tx := range txs {
+	start := 0
+	end := 0
+	for end < len(txs) {
+		tx := txs[end]
 		inputs := tx.InputIDs()
-		overlaps := consumed.Overlaps(inputs) // See if this tx shares inputs with another one in this batch
-		if len(batch) >= t.Params.BatchSize || (force && overlaps) {
-			// The batch is big enough to issue, or we need to issue this batch
-			// because we're forcing each tx to be issued but adding [tx] to
-			// this batch would result in a vertex with conflicting txs.
-			if err := t.issueBatch(batch); err != nil {
+		overlaps := consumed.Overlaps(inputs)
+		if end-start >= t.Params.BatchSize || (force && overlaps) {
+			if err := t.issueBatch(txs[start:end]); err != nil {
 				return err
 			}
-			batch = make([]snowstorm.Tx, 0, t.Params.BatchSize)
+			start = end
 			consumed.Clear()
 			issued = true
 			overlaps = false
 		}
+
 		if txID := tx.ID(); !overlaps && // should never allow conflicting txs in the same vertex
 			!issuedTxs.Contains(txID) && // shouldn't issue duplicated transactions to the same vertex
 			(force || t.Consensus.IsVirtuous(tx)) && // force allows for a conflict to be issued
 			(!t.Consensus.TxIssued(tx) || orphans.Contains(txID)) { // should only reissue orphaned txs
-			batch = append(batch, tx)
+			end++
 			issuedTxs.Add(txID)
 			consumed.Union(inputs)
+		} else {
+			newLen := len(txs) - 1
+			txs[end] = txs[newLen]
+			txs[newLen] = nil
```

Comment:
This line doesn't do anything

Reply:
Isn't this required to avoid a potential memory leak?

Reply:
By nilling it out you may let the GC potentially collect the tx earlier... But it certainly isn't needed to prevent a memory leak (the tx will be collected once the array is collected)

Reply:
We'll keep it